hydra-cluster/test/Test/EndToEndSpec.hs

```diff
@@ -495,6 +511,46 @@ spec = around (showLogsOnFailure "EndToEndSpec") $ do
             logfile <- readFileBS logFilePath
             BS.length logfile `shouldSatisfy` (> 0)
 
+    describe "forking eras" $ do
+      it "does report on unsupported era" $ \tracer -> do
+        withClusterTempDir "unsupported-era" $ \tmpDir -> do
+          args <- setupCardanoDevnet tmpDir
+          forkIntoConwayInEpoch tmpDir args 1
+          withCardanoNode (contramap FromCardanoNode tracer) defaultNetworkId tmpDir args $ \node@RunningNode{nodeSocket} -> do
+            hydraScriptsTxId <- publishHydraScriptsAs node Faucet
+            chainConfig <- chainConfigFor Alice tmpDir nodeSocket [] cperiod
+            withHydraNode' chainConfig tmpDir 1 aliceSk [] [1] hydraScriptsTxId Nothing $ \out err ph -> do
+              -- Assert nominal startup
+              waitForLog 5 out "missing NodeOptions" (Text.isInfixOf "NodeOptions")
+
+              delayEpoch tmpDir args 1
+
+              waitForProcess ph `shouldReturn` ExitFailure 1
+              hGetContents err >>= (`shouldContain` "Received blocks in not supported era.")
+
+      it "does report on unsupported era on startup" $ \tracer -> do
+        withClusterTempDir "unsupported-era-startup" $ \tmpDir -> do
+          args <- setupCardanoDevnet tmpDir
+          forkIntoConwayInEpoch tmpDir args 1
+          withCardanoNode (contramap FromCardanoNode tracer) defaultNetworkId tmpDir args $ \node@RunningNode{nodeSocket} -> do
+            hydraScriptsTxId <- publishHydraScriptsAs node Faucet
+            chainConfig <- chainConfigFor Alice tmpDir nodeSocket [] cperiod
+
+            delayEpoch tmpDir args 2
+
+            withHydraNode' chainConfig tmpDir 1 aliceSk [] [1] hydraScriptsTxId Nothing $ \_out err ph -> do
+              waitForProcess ph `shouldReturn` ExitFailure 1
+              hGetContents err >>= (`shouldContain` "Connected to cardano-node in not supported era.")
+
+-- | Wait for given number of epochs. This uses the epoch and slot lengths from
+-- the 'ShelleyGenesisFile' of the node args passed in.
+delayEpoch :: FilePath -> CardanoNodeArgs -> Natural -> IO ()
+delayEpoch stateDirectory args epochs = do
+  shellyGenesisFile :: Aeson.Value <- unsafeDecodeJsonFile (stateDirectory </> nodeShelleyGenesisFile args)
+  let slotLength :: Double = shellyGenesisFile & getField "slotLength"
+      epochLength = shellyGenesisFile & getField "epochLength"
```

Comment:
Could: simplify this by using `lens-aeson` (which used by `getField`)

I think this would work here:
```suggestion
      epochLength = shellyGenesisFile ^. key "epochLength"
```

If it does, no need to add `getField`

Reply:
unfortunately this suggestions does not work out of the box.

this function gives me back a Value and I still need to parse it and pattern match on.

what am i missing?